apps/game/models/challenge.py

```diff
@@ -0,0 +1,95 @@
+from django.contrib.auth.models import User
+from .game import Game
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from apps.accounts.models import Team
+
+
+class Challenge(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.CharField(max_length=2047)
+    start_time = models.DateTimeField()
+    end_time = models.DateTimeField()
+    registration_start_time = models.DateTimeField()
+    registration_end_time = models.DateTimeField()
+    registration_open = models.BooleanField()
+    team_size = models.IntegerField()
+    entrance_price = models.IntegerField()  # In Toomans, 0 for free
+    game = models.ForeignKey(Game)
+
+    def can_register(self):
+        return self.registration_open #and (current time between reg_start_time and reg_end_time)
+
+    def open_registration(self):
+        self.registration_open = True
+        self.save()
+
+    def close_registration(self):
+        self.registration_open = False
+        self.save()
+
+
+class TeamParticipatesChallenge(models.Model):
+    team = models.ForeignKey(Team, related_name='challanges')
+    challenge = models.ForeignKey(Challenge, related_name='teams')
+
+    def all_members_accepted(self):
+        """
+        :rtype: bool
+        """
+        users = self.team.participants.all()
+        ok = True
+        for user in users:
+            ok &= UserAcceptsTeamInChallenge.objects.filter(team=self.team, user=user).exists()
+        return ok
+
+    def has_payed(self):
+        """
+        :rtype: bool
+        """
+        pass
+
+    def get_final_submission(self):
+        """
+        :rtype: TeamSubmission
+        """
+        try:
+            return TeamSubmission.objects.filter(team=self, is_final=True).first()
```

Comment:
Didn't we decide to put a foreign key from `TeamParticipatesInCompetition`  or `TeamParticipatesInChallenge` to the final submission? 
Otherwise assert if the query only returns one submission.